concrete-csprng/src/seeders/mod.rs

```diff
@@ -0,0 +1,43 @@
+//! A module containing seeders objects.
+//!
+//! When initializing a generator, one needs to provide a [`Seed`], which is then used as key to the
+//! AES blockcipher. As a consequence, the quality of the outputs of the generator is directly
+//! conditioned by the quality of this seed. This module proposes different mechanisms to deliver
+//! seeds that can accomodate varying scenarios.
+
+/// A seed value, used to initialize a generator.
+#[derive(Debug, Copy, Clone)]
+pub struct Seed(pub u128);
+
+/// A trait representing a seeding strategy.
+pub trait Seeder {
+    /// Generates a new seed.
+    fn seed(&mut self) -> Seed;
+}
+
+mod implem;
+pub use implem::*;
+
+#[cfg(test)]
+mod generic_tests {
+    use crate::seeders::Seeder;
+
+    /// Naively verifies that two fixed-size sequences generated by repeatedly calling the seeder
+    /// are different.
+    #[allow(unused)]
+    pub fn check_seeder_fixed_sequences_different<S: Seeder, F: Fn(usize) -> S>(
```

Comment:
just wondering about the choice of usize here as the LinuxSeeder converts directly to a u128, is it better to have a Fn(u128) -> S so that we only have to care about the conversion in the function below ? (passing i as u128) and avoids having to do that in other potential implementation of seeders ?

Reply:
Done !